pyOCD/flash/flash.py

```diff
@@ -105,20 +110,42 @@ def __init__(self, target, flash_algo):
     def minimumProgramLength(self):
         return self.min_program_length
 
-    def init(self):
+    def init(self, adr=None, clk=0, fnc=None):
         """
         Download the flash algorithm in RAM
         """
+        if adr is None:
```

Comment:
what cases do you anticipate for the `adr` and `clk` to be passed explicitly?
if none - please consider removing them